src/main/java/org/jenkinsci/test/acceptance/po/DumbSlave.java

```diff
@@ -48,9 +50,14 @@ public void asLocal() {
         File jar = new File(tmpDir, "slave"+createRandomName()+".jar");
         String command;
         if (SystemUtils.IS_OS_UNIX) {
+            String java = "java";
+            String jh = System.getProperty("java.home");
+            if (!StringUtils.isEmpty(jh)) {
+                java = new File(jh, "bin/java").getAbsolutePath();
```

Comment:
I was just pointing out that it would be safer to write

```java
File javaBin = new File(jh, "bin/java");
if (javaBin.isExecutable()) {
    java = javaBin.getAbsolutePath();
}
```